### PR TITLE
Remove dependency on QUdpSocket in UdpDataProtocol

### PR DESCRIPTION
**Pull request by psiborg112**
Only use bsd socket and winsock functions instead of the current mix of these and QUdpSocket. (Some bad interaction between them seems to be causing issues on windows.)

### CODE REVIEW

**Comment by psiborg112** [comment]
Meant to add that this seems to resolve #117.

File: src/UdpDataProtocol.cpp [hunk 1]
```diff
@@ -358,15 +369,24 @@ int UdpDataProtocol::sendPacket(const char* buf, const size_t n)
...
+    struct sockaddr_storage addr;
+    std::memset(&addr, 0, sizeof(addr));
+    socklen_t sa_len = sizeof(addr);
+    ::recvfrom(mSocket, buf, 1, 0, (struct sockaddr*) &addr, &sa_len);
+    peerHostAddress.setAddress((struct sockaddr*) &addr);
+    if (mIPv6) {
+        port = ((struct sockaddr_in*) &addr)->sin_port;
+    } else {
+        port = ((struct sockaddr_in6*) &addr)->sin6_port;
+    }
```
Comment (ntonnaett) [inline src/UdpDataProtocol.cpp:389]: Is this the right way around?

Comment (psiborg112) [reply src/UdpDataProtocol.cpp:389]: It is not - good catch.

File: src/UdpDataProtocol.cpp [hunk 2]
```diff
@@ -507,18 +525,17 @@ void UdpDataProtocol::run()
...
         }
-        int first_packet_size = UdpSocket.pendingDatagramSize();
         // The following line is the same as
-        int8_t* first_packet = new int8_t[first_packet_size];
+        int8_t* first_packet = new int8_t[full_redundant_packet_size];
         /// \todo fix this to avoid memory leaks
         // but avoids memory leaks
         //std::tr1::shared_ptr<int8_t> first_packet(new int8_t[first_packet_size]);
-        receivePacket( UdpSocket, reinterpret_cast<char*>(first_packet), first_packet_size);
+        receivePacket( reinterpret_cast<char*>(first_packet), full_redundant_packet_size);
```
Comment (ntonnaett) [inline src/UdpDataProtocol.cpp:538]: I thought the first packet is just the handshake. Why do we need full redundant packet size at this point? Actually those packet size names are not clear. But that's for another day.

Comment (psiborg112) [reply src/UdpDataProtocol.cpp:538]: We don't. Technically we could just use the header size and let the rest of the packet go, but reading the header size would have meant adding an additional getter in the JackTrip class. And this way we can just allocate a single buffer for both the initial packet and all of the subsequent packets.

Comment (psiborg112) [reply src/UdpDataProtocol.cpp:538]: (It's worth noting that the code that this replaces used to read the full first packet into the buffer as well, so this is no less efficient than the previous code. And the first packet isn't any smaller than the packets that follow it - there's never a header sent by itself without audio data.)

**Comment by cchafe** [comment]
The new idea is to leave "main" pristine and equal to the stable release which has a tag (in this case, v1.2.1)
Want to merge it to "dev" instead?

**Comment by psiborg112** [comment]
Happy to merge to dev, but I think we should try to test this as much as possible and shift it to main quickly because it's a pretty bad bug. I think it's a hangover from the previous issues we were having, and that our workaround only works for certain qt and mingw versions. (If it helps, I recently ran a hub server and one of the hub clients in a 24 hour Berlin/Perth/Melbourne livestream on this code without issue.)

**Comment by psiborg112** [comment]
I think I need to close this to merge it to dev, so will do that.

**Pull request closed without merging**